Engine/ac/global_audio.cpp

```diff
@@ -386,21 +411,26 @@ void PlaySilentMIDI (int mnum) {
     play.silent_midi = mnum;
     play.silent_midi_channel = SCHAN_SPEECH;
     stop_and_destroy_channel(play.silent_midi_channel);
-    channels[play.silent_midi_channel] = load_sound_clip_from_old_style_number(true, mnum, false);
-    if (channels[play.silent_midi_channel] == NULL)
+
+    AudioChannelsLock lock;
+    auto* ch = lock.SetChannel(play.silent_midi_channel,load_sound_clip_from_old_style_number(true, mnum, false));
```

Comment:
Should you do the same thing here? Not overwrite the channel on failure? I suppose we quit() on failure this time.

Reply:
It seems like channels were usually overwritten with null on failure. I don't know if this is requirement so tried to keep same logic.

As for quitting, this is very inconsistent in AGS. I recieved requests to replace them with warnings to the log on numerous occasions. IMHO this should be configurable, in other words an "assert"-like function that either prints into log or quits depending on setup. Then we may send a message to the IDE if it's connected in debug session too, and even break at this point.